Flex test suite is failing in master

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The flex test suite is failing in master since the merge of #6317. 

For future reference, we need to run with [test full] whenever making broad changes to the codebase so it is more obvious that something isn't an intermittent failure.

```
--- FAIL: TestCephFlexSuite (538.19s)
    --- PASS: TestCephFlexSuite/TestBlockStorageMountUnMountForDifferentAccessModes (150.21s)
    --- FAIL: TestCephFlexSuite/TestBlockStorageMountUnMountForStatefulSets (85.39s)
        ceph_flex_test.go:170: 
            	Error Trace:	ceph_flex_test.go:170
            	            				ceph_flex_test.go:163
            	Error:      	Received unexpected error:
            	            	services "block-stateful-set" not found
            	Test:       	TestCephFlexSuite/TestBlockStorageMountUnMountForStatefulSets
        ceph_flex_test.go:172: 
            	Error Trace:	ceph_flex_test.go:172
            	            				ceph_flex_test.go:163
            	Error:      	Received unexpected error:
            	            	statefulsets.apps "ststest" not found
            	Test:       	TestCephFlexSuite/TestBlockStorageMountUnMountForStatefulSets
```

**Expected behavior:**
Integration tests should pass in master.

**How to reproduce it (minimal and precise):**
<!-- Please let us know any circumstances for reproduction of your bug. -->
Run the flex test suite.

--- a/tests/integration/ceph_base_file_test.go
+++ b/tests/integration/ceph_base_file_test.go
@@ -329,7 +329,7 @@ func cleanupFilesystemConsumer(helper *clients.TestClient, k8sh *utils.K8sHelper
 		assert.Fail(s.T(), fmt.Sprintf("make sure %s pod is terminated", podName))
 	}
 	err = helper.FSClient.DeletePVC(namespace, podName)
-	assert.NoError(s.T(), err)
+	assertNoErrorUnlessNotFound(s, err)
 	logger.Infof("File system consumer deleted")
 }
 

--- a/tests/integration/ceph_flex_test.go
+++ b/tests/integration/ceph_flex_test.go
@@ -167,9 +167,9 @@ func (s *CephFlexDriverSuite) statefulSetDataCleanup(poolName, storageClassName,
 	listOpts := metav1.ListOptions{LabelSelector: "app=" + statefulSetName}
 	// Delete stateful set
 	err := s.kh.Clientset.CoreV1().Services(defaultNamespace).Delete(statefulSetName, &delOpts)
-	assert.NoError(s.T(), err)
+	assertNoErrorUnlessNotFound(s.Suite, err)
 	err = s.kh.Clientset.AppsV1().StatefulSets(defaultNamespace).Delete(statefulPodsName, &delOpts)
-	assert.NoError(s.T(), err)
+	assertNoErrorUnlessNotFound(s.Suite, err)
 	err = s.kh.Clientset.CoreV1().Pods(defaultNamespace).DeleteCollection(&delOpts, listOpts)
 	assert.NoError(s.T(), err)
 
@@ -236,9 +236,9 @@ func (s *CephFlexDriverSuite) TearDownSuite() {
 		"rwo-block-ro-two", "rwx-block-rw-one", "rwx-block-rw-two", "rwx-block-ro-one", "rwx-block-ro-two")
 	assert.NoError(s.T(), err)
 	err = s.testClient.BlockClient.DeletePVC(s.namespace, s.pvcNameRWO)
-	assert.NoError(s.T(), err)
+	assertNoErrorUnlessNotFound(s.Suite, err)
 	err = s.testClient.BlockClient.DeletePVC(s.namespace, s.pvcNameRWX)
-	assert.NoError(s.T(), err)
+	assertNoErrorUnlessNotFound(s.Suite, err)
 	err = s.testClient.BlockClient.DeleteStorageClass("rook-ceph-block-rwo")
 	assert.NoError(s.T(), err)
 	err = s.testClient.BlockClient.DeleteStorageClass("rook-ceph-block-rwx")

